tests/Resources/Page/ODH/JupyterHub/JupyterLabLauncher.robot

```diff
@@ -378,6 +379,12 @@ Check Versions In JupyterLab
               ${return_status} =    Set Variable    FAIL
             END
         END
+        Continue For Loop If  "${libDetail}[0]" not in ${packages}
+        Run Keyword Unless    "${libDetail}[0]" in ${package_versions}
+        ...    Set To Dictionary    ${package_versions}    ${libDetail}[0]=${libDetail}[1]
+        IF    "${package_versions["${libDetail}[0]"]}" != "${libDetail}[1]"
+             ${return_status} =    Set Variable    FAIL
+        END
```

Comment:
I'm not getting what this code is achieving, could you pls explain?

Reply:
`${libDetail[0]}` is the name of library `${libDetail[1]}` is the current version of that library. I am checking if the current version of library is matching the version of same library stored in dictionary(dictionary contains versions after spawning multiple notebooks)

Reply:
`Run Keyword Unless    "${libDetail}[0]" in ${package_versions}
        ...    Set To Dictionary    ${package_versions}    ${libDetail}[0]=${libDetail}[1]
`
Updates the `package_versions` after spawning notebook if that library is not present in it.

Reply:
yeah that was clear..I didn't get the goal but you explained me offline.

However, I have two more comments on this:
1. are you sure `${package_versions}` is being updated? since it is passed as argument, I fear it is used as local variable in the kw and not as global. Besides, the `Run Keyword Unless` seems to be always "green" in the logs
2. `Run Keyword Unless` has been deprecated in Robotframework 5.0 which we are going use in few time. So, could you replace it?

Reply:
Updated.